microsoftgraph/docs-conceptual/tutorial-grant-delegated-api-permissions.md

```diff
@@ -0,0 +1,166 @@
+UserConsentDescription  : Allows the app to read and write the full set of profile properties, reports, and managers of other users in your organization, on your behalf.
+UserConsentDisplayName  : Read and write all users' full profiles
+Value                   : User.ReadWrite.All
+AdditionalProperties    : {}
+
+AdminConsentDescription : Allows the app to list groups, and to read their properties and all group memberships on behalf of the signed-in user.  Also allows the app to read calendar, conversations, files,
+                          and other group content for all groups the signed-in user can access.
+AdminConsentDisplayName : Read all groups
+Id                      : 5f8c59db-677d-491f-a6b8-5f174b11ec1d
+IsEnabled               : True
+Origin                  :
+Type                    : Admin
+UserConsentDescription  : Allows the app to list groups, and to read their properties and all group memberships on your behalf.  Also allows the app to read calendar, conversations, files, and other group
+                          content for all groups you can access.
+UserConsentDisplayName  : Read all groups
+Value                   : Group.Read.All
+AdditionalProperties    : {}
+```
+
+## Step 2: Create a client service principal
+
+The first step in granting consent is to [create the service principal](/powershell/module/microsoft.graph.applications/new-mgserviceprincipal?view=graph-powershell-1.0&preserve-view=true). To do so, you'll need the `App Id` of your application.
+
+### Register an application with Azure AD
+
+If the application is not available, register an application with Azure AD.
+
+```powershell
+New-MgApplication -DisplayName 'My application' | 
+  Format-List Id, DisplayName, AppId, SignInAudience, PublisherDomain
+```
+
+```Output
+Id              : 40cbfad6-f138-4fb4-9e7f-5a44044efbd6
+DisplayName     : My application
+AppId           : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience  : AzureADandPersonalMicrosoftAccount
+PublisherDomain : Contoso.com
+```
+
+### Create a service principal for the application
+
+```powershell
+New-MgServicePrincipal -AppId '05210c44-437f-4a40-bd38-b5b4eaf251ef' | 
+  Format-List Id, DisplayName, AppId, SignInAudience
+```
+
+```Output
+Id             : 22c1770d-30df-49e7-a763-f39d2ef9b369
+DisplayName    : My application
+AppId          : 05210c44-437f-4a40-bd38-b5b4eaf251ef
+SignInAudience : AzureADandPersonalMicrosoftAccount
+```
+
+## Step 3: Grant delegated permission to the client service principal
+
+To create a delegated permission grant, you'll need the following information:
+
+1. **ClientId** - object Id of the client service principal authorized to act on behalf of the user. In this case, the service principal we created in step 2.
+1. **ConsentType** - `AllPrincipals` to authorize all users in the tenant or `Principal` for a single user.
+1. **PrincipalId** - Id of the user for *Principal* consents, `null` for *AllPrincipals* consents.
```

Comment:
Suggest reversing the order of this line to match the line right above - AllPrincipals, then Principal.